python-app/docker/action.yml

```diff
@@ -0,0 +1,61 @@
+name: Build
+description: Build and publish image
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Check if step is needed
+      run: |
+        echo docker_needed=${{ startsWith(github.ref, 'refs/tags/') || startsWith(github.ref, 'refs/heads/feature/') || github.ref == 'main' }} >> $GITHUB_ENV
+      shell: bash
+
+    - name: Clone
+      uses: actions/checkout@v2
+      if: ${{ env.docker_needed == 'true' }}
+
+    - name: Build image name (lowercase)
+      run: |
+        echo IMAGE="ghcr.io/${GITHUB_REPOSITORY,,}" >> $GITHUB_ENV
+      shell: bash
+
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v3
+      with:
+        images: ${{ env.IMAGE }}
+        flavor: |
+            latest=false
+        tags: |
+            type=raw,value=latest,enable=${{ github.ref == 'refs/heads/main' }}
+            type=ref,event=branch
+            type=ref,event=tag
+            type=ref,event=pr
+      if: ${{ env.docker_needed == 'true' }}
```

Comment:
Personally I prefer when the `if` is as high up as possible in the list, as it makes reading active steps easier.